CMakeLists.txt

```diff
@@ -175,7 +175,9 @@ add_compile_options(
     # The following list is not indicative of what we _desire_ only the status quo to build.
     -Wno-deprecated-declarations
     -Wno-pessimizing-move
-    -Wno-unused-but-set-variable
+    # Vars that are used only in Asserts will appear as dead stores in builds
+    # where the preprocessor has stripped away the assert.
+    "$<$<CONFIG:Release>:-Wno-unused-but-set-variable>"
```

Comment:
Considering CI only runs in release, and that the default build type for `build_metal.sh` is also release, I suspect it will be too easy to break debug build once this gets merged. Arguments that only get passed to functions for `TT_ASSERT` are a code smell and should be removed anyway. I would prefer for that to be done or for this to remain unconditionally disabled.

Reply:
yeah, I'm okay with that. I wasn't aware we had non-release PR gates enabled now. Thank you for pointing that out.